### ISSUE TITLE
Boolean datatype not handled with MySQL dialect

### ISSUE THREAD

**Issue by scampi** [jOOQ/jOOQ#4079]
Using Jooq v3.5.2 and H2 in memory as the database in MYSQL mode.

The snippet below throws the following exception:

```
org.jooq.exception.DataAccessException: SQL [create table `table`(`boolean` unsigned null)]; Unknown data type: "UNSIGNED"; SQL statement:
create table `table`(`boolean` unsigned null) [50004-185]
	at org.jooq.impl.Utils.translate(Utils.java:1553)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:609)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:351)
```

Here is how the table is created:

```java
Class.forName("org.h2.Driver");
Connection conn = DriverManager.getConnection("jdbc:h2:mem:testdb;MODE=MYSQL");
DSLContext create = DSL.using(conn, SQLDialect.MYSQL);

create.createTable("table")
         .column("bool", SQLDataType.BOOLEAN)
         .execute();
```

**Comment by lukaseder** [comment]
Thanks for reporting. This is a known issue: #3341. The fix will be published soon, also in 3.5.3

**Comment by lukaseder** [comment]
Interesting approach:

```java
Connection conn = DriverManager.getConnection("jdbc:h2:mem:testdb;MODE=MYSQL");
DSLContext create = DSL.using(conn, SQLDialect.MYSQL);
```

We currently don't run our integration tests this way, but it would be interesting to do so. I suspect we'll be able to report quite a few bugs to H2, to help them improve...
